src/index.js

```diff
@@ -202,172 +297,236 @@ class PeerId {
-  const privDigest = await computeDigest(privKey.public)
+  /**
+   * Create PeerId from private key.
+   *
+   * @param {Uint8Array | string} key - Private key, as Uint8Array or base64-encoded string.
+   * @returns {Promise<PeerId>}
+   */
+  static async createFromPrivKey (key) {
+    if (typeof key === 'string') {
+      key = uint8ArrayFromString(key, 'base64pad')
+    }
 
-  let pubDigest
+    if (!(key instanceof Uint8Array)) {
+      throw new Error('Supplied key is neither a base64 string nor a Uint8Array')
+    }
 
-  if (pub) {
-    pubDigest = await computeDigest(pub)
+    const privKey = await cryptoKeys.unmarshalPrivateKey(key)
+    return computePeerId(privKey.public, privKey)
   }
 
-  if (pub && !uint8ArrayEquals(privDigest, pubDigest)) {
-    throw new Error('Public and private key do not match')
-  }
+  /**
+   * Create PeerId from PeerId JSON formatted object.
+   *
+   * @param {PeerIdJSON} obj
+   * @returns {Promise<PeerId>}
+   */
+  static async createFromJSON (obj) {
+    const id = mh.fromB58String(obj.id)
+    const rawPrivKey = obj.privKey && uint8ArrayFromString(obj.privKey, 'base64pad')
+    const rawPubKey = obj.pubKey && uint8ArrayFromString(obj.pubKey, 'base64pad')
+    const pub = rawPubKey && await cryptoKeys.unmarshalPublicKey(rawPubKey)
+
+    if (!rawPrivKey) {
+      return new PeerId(id, undefined, pub)
+    }
 
-  if (id && !uint8ArrayEquals(privDigest, id)) {
-    throw new Error('Id and private key do not match')
-  }
+    const privKey = await cryptoKeys.unmarshalPrivateKey(rawPrivKey)
+    const privDigest = await computeDigest(privKey.public)
 
-  return new PeerIdWithIs(id, privKey, pub)
-}
+    if (pub) {
+      const pubDigest = await computeDigest(pub)
 
-exports.createFromProtobuf = async (buf) => {
-  if (typeof buf === 'string') {
-    buf = uint8ArrayFromString(buf, 'base16')
+      if (!uint8ArrayEquals(privDigest, pubDigest)) {
+        throw new Error('Public and private key do not match')
+      }
+    }
+
+    if (id && !uint8ArrayEquals(privDigest, id)) {
+      throw new Error('Id and private key do not match')
+    }
+
+    return new PeerId(id, privKey, pub)
   }
 
-  let { id, privKey, pubKey } = PeerIdProto.decode(buf)
+  /**
+   * Create PeerId from Protobuf bytes.
+   *
+   * @param {Uint8Array | string} buf - Protobuf bytes, as Uint8Array or hex-encoded string.
+   * @returns {Promise<PeerId>}
+   */
+  static async createFromProtobuf (buf) {
+    if (typeof buf === 'string') {
+      buf = uint8ArrayFromString(buf, 'base16')
+    }
 
-  privKey = privKey ? await cryptoKeys.unmarshalPrivateKey(privKey) : false
-  pubKey = pubKey ? await cryptoKeys.unmarshalPublicKey(pubKey) : false
+    const { id, privKey: privKeyBytes, pubKey: pubKeyBytes } = PeerIdProto.decode(buf)
 
-  let pubDigest
-  let privDigest
+    /**
+     * @type {PrivateKey | false}
+     */
+    const privKey = privKeyBytes ? await cryptoKeys.unmarshalPrivateKey(privKeyBytes) : false
```

Comment:
probably simpler here and below to just

```js
const privKey = privKeyBytes && await cryptoKeys.unmarshalPrivateKey(privKeyBytes)
```

The type would change to `| undefined `